match/lib/match/importer.rb

```diff
@@ -125,13 +140,46 @@ def import_cert(params, cert_path: nil, p12_path: nil, profile_path: nil)
       storage.clear_changes if storage
     end
 
-    def ensure_valid_file_path(file_path, file_description, file_extension, optional: false)
+    def resolve_inputs(params, cert_path, p12_path, profile_path)
+      cert_path ||= params[:import_certificate_file_path]
+      p12_path ||= params[:import_certificate_private_key_file_path]
+      profile_path ||= params[:import_provisioning_profile_file_path]
+
+      # Get and verify cert, p12 and profiles path
+      if cert_path.nil?
+        cert_path = ensure_valid_file_path(cert_path, "Certificate", ".cer", optional: true, skip_prompt: params[:import_suppress_ui_file_path_prompts])
+      else
+        cert_path = ensure_valid_file_path(cert_path, "Certificate", ".cer", optional: true, skip_prompt: true)
+      end
+
+      if p12_path.nil?
+        p12_path = ensure_valid_file_path(p12_path, "Private key", ".p12", optional: true, skip_prompt: params[:import_suppress_ui_file_path_prompts])
+      else
+        p12_path = ensure_valid_file_path(p12_path, "Private key", ".p12", optional: true, skip_prompt: true)
+      end
+
+      if profile_path.nil?
+        profile_path = ensure_valid_file_path(profile_path, "Provisioning profile", ".mobileprovision or .provisionprofile", optional: true, skip_prompt: params[:import_suppress_ui_file_path_prompts])
+      else
+        profile_path = ensure_valid_file_path(profile_path, "Provisioning profile", ".mobileprovision or .provisionprofile", optional: true, skip_prompt: true)
+      end
+
+      return [cert_path, p12_path, profile_path]
+    end
+
+    def ensure_valid_file_path(file_path, file_description, file_extension, optional: false, skip_prompt: false)
       optional_file_message = optional ? " or leave empty to skip this file" : ""
-      file_path ||= UI.input("#{file_description} (#{file_extension}) path#{optional_file_message}:")
+      if file_path.nil? && !skip_prompt
+        file_path ||= UI.input("#{file_description} (#{file_extension}) path#{optional_file_message}:")
+      end
+      raw_file_path = file_path
+
+      if file_path
+        file_path = File.absolute_path(file_path) unless file_path == ""
```

Comment:
Would it be useful to also use `File.expand_path` here to support ~/ based path?

Reply:
addressed in 407c004b3ab82b308976d24c231ef43b8f0f5cd6